src/xpub.cpp

```diff
@@ -280,6 +287,8 @@ int zmq::xpub_t::xrecv (msg_t *msg_)
     // set metadata only if there is some
     if (metadata_t* metadata = pending_metadata.front ()) {
         msg_->set_metadata (metadata);
+        // Remove ref corresponding to vector placement
+        metadata->drop_ref();
```

Comment:
Upon further review this probably warrants a conditional LIBZMQ_DELETE, although the preceding set_metadata should also have incremented the refcount